crates/noirc_evaluator/src/brillig/brillig_gen.rs

```diff
@@ -146,10 +161,44 @@ impl BrilligGen {
                 let result_register = self.get_or_create_register(result_ids[0]);
                 self.convert_ssa_binary(binary, dfg, result_register);
             }
-            _ => todo!("ICE: Instruction not supported"),
+            Instruction::Not(value) => {
+                let result_ids = dfg.instruction_results(instruction_id);
+                let result_register = self.get_or_create_register(result_ids[0]);
+
+                assert_eq!(
```

Comment:
It possible to do what some languages do and make NOT be a bitwise not operator, taking the twos complement. I did it this way because in Acir gen, we also only allow booleans for the not operator